### PR TITLE
Fix deprecated code usage

### LINKED ISSUE #30147: PHP Deprecated
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Some php code used triggers deprecated warnings in the php logs.
```
1.
[28-Oct-2022 10:04:41 Europe/Stockholm] PHP Deprecated:  strlen(): Passing null to parameter #1 ($string) of type string is deprecated in D:\wamp64\www\prestashop800\vendor\twig\twig\src\Loader\FilesystemLoader.php on line 134

2.
[28-Oct-2022 10:04:41 Europe/Stockholm] PHP Deprecated:  strncmp(): Passing null to parameter #1 ($string1) of type string is deprecated in D:\wamp64\www\prestashop800\vendor\twig\twig\src\Loader\FilesystemLoader.php on line 135

3.
[28-Oct-2022 13:35:34 Europe/Stockholm] PHP Deprecated:  ucfirst(): Passing null to parameter #1 ($string) of type string is deprecated in D:\wamp64\www\prestashop800\classes\controller\AdminController.php on line 983

4.
[28-Oct-2022 09:56:16 Europe/Stockholm] PHP Deprecated:  uasort(): Returning bool from comparison function is deprecated, return an integer less than, equal to, or greater than zero in D:\wamp64\www\prestashop800\controllers\admin\AdminModulesController.php on line 110

```

### Expected behavior

No entries in the log when using the software.

### Steps to reproduce

0. Make sure debug mode is turned off in performance settings.
1. Go to orders controller in admin
2. log entries marked 1 and 2 should be in the php logs.
3. Go to modules tab, click on configure on a module like ps_brandslist
4. log entry marked 3 will be in the logs.
5. The last entry marked 4 is a bit harder as it requires a module with an admincontroller where the action parameter is not set.

### PrestaShop version(s) where the bug happened

8.0.0

### PHP version(s) where the bug happened

8.1.12

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by Prestaworks**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Fix some deprecated warnings
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | yno
| Deprecations?     | yes
| Fixed ticket?     | Fixes #30147.  (Not the twig warnings, but the other two atleast)
| Related PRs       | 
| How to test?      | Follow steps in issue.
| Possible impacts? |

### CODE REVIEW

File: controllers/admin/AdminModulesController.php
Comment (lartist) [inline controllers/admin/AdminModulesController.php]: Hi @Prestaworks If you want to fit the old logic you should use instead:
```php
return 1
```

File: classes/controller/AdminController.php
Comment (FabienPapet) [inline classes/controller/AdminController.php:340]: This looks like a BC break for me. The best fix for me is to convert null values before calling strcmp function

File: controllers/admin/AdminModulesController.php
Comment (lartist) [inline controllers/admin/AdminModulesController.php]: I think there is a misunderstanding here.
If you see the following snippet you'll see that the old logic is not the same as the new:
```php

(bool) ('d' > 'abcd')	//=> true

strlen('d') <=> strlen('abcd')	//=> -1

```
You just need to remove `strlen` as before it was not the lengths which was compared but the `>` comparator on string proceed as an alphabetic order comparison. So I think the following code is more appropriate
```php
'd' <=> 'abcd'	//=> 1
```

File: controllers/admin/AdminModulesController.php
Comment (FabienPapet) [inline controllers/admin/AdminModulesController.php]: This is also a BC break, as this function was returning boolean values, and is now returning integer.

**Review by FabienPapet — DISMISSED** [review]
Hello, 

Thank you for your pull request. There are some changes needed before we can merge your work.

If the bug you encountered was in PS 8.0, you can rebase onto the `8.0.x` branch.

File: controllers/admin/AdminModulesController.php
Comment (Prestaworks) [reply controllers/admin/AdminModulesController.php]: https://www.php.net/manual/en/function.uasort
actually if the values are equal, a zero is expected.
"The comparison function must return an integer less than, equal to, or greater than zero if the first argument is considered to be respectively less than, equal to, or greater than the second. "

Comment (Prestaworks) [reply controllers/admin/AdminModulesController.php]: that's the whole point as the uasort that calls this function expects an integer and not a boolean.

File: controllers/admin/AdminModulesController.php
Comment (Prestaworks) [reply controllers/admin/AdminModulesController.php]: makes sense :) I'll change that

File: classes/controller/AdminController.php
Comment (Prestaworks) [reply classes/controller/AdminController.php:340]: I can't really see how that is a BC break.

File: controllers/admin/AdminModulesController.php
Comment (lartist) [reply controllers/admin/AdminModulesController.php]: Yes you're right and here the exepected behavior is to consider than "Other Modules" should always be first, then you always have to return 1 otherwise it will be compared as equal than others.

File: classes/controller/AdminController.php
Comment (FabienPapet) [reply classes/controller/AdminController.php:340]: Changing a default value **is** a BC break. If you want to force a null value to be a string, you can add `(string) $this->action` before

File: controllers/admin/AdminModulesController.php
Comment (FabienPapet) [reply controllers/admin/AdminModulesController.php]: yes, but this function is returning a boolean, and cannot return now a integer (because of BC Break). The deprecation comes from the function usage, not the function itself. There would be no problem if the function was private, but as it is public, we must protect usages for other developers using that function for other purposes.

**Comment by lmeyer1** [comment]
@FabienPapet What hinders this PR to be merged? It is an appropriate fix.

**Comment by Prestaworks** [comment]
Maybe it can be merged into PS 9, major versions should allow BC breaks right? (if the issue still exists on PS9)

File: controllers/admin/AdminModulesController.php
Comment (lmeyer1) [reply controllers/admin/AdminModulesController.php]: @FabienPapet
Is such a strict BC break policiy appropriate here?
If we cannot merge this PR because of a BC break, PS 8.0 does not support PHP 8.0. If PS 8.0 is supposed to support PHP 8.0, we must merge this PR as soon as possible.

IMHO, this is a issue that was missed when PS was made compatible with PHP 8.0. Or am I missing something?

Comment (FabienPapet) [reply controllers/admin/AdminModulesController.php]: it was at the time the pr was written, because of the bc break promise for 8.1 . Now that the next version is 9.0, such a change should be allowed.

Comment (FabienPapet) [reply controllers/admin/AdminModulesController.php]: Just don't forget to rebase your PR ;)

Comment (lmeyer1) [reply controllers/admin/AdminModulesController.php]: @Prestaworks 
Can you fix this line to `return 1;`?
If I understand well this is the only thing that is keeping this PR from being merged.

**Review by lmeyer1 — CHANGES_REQUESTED** [review]

**Review by davidglezz — APPROVED** [review]

**Comment by Prestaworks** [comment]
These files are not used any more right? it's all symfony controllers in BO in the latest versions?

File: controllers/admin/AdminModulesController.php
Comment (davidglezz) [inline controllers/admin/AdminModulesController.php:124]: Aside, `check` verb and plurals for comparator function is inusual right?
```suggestion
    public function checkCategoriesNames($a, $b) {
        // ADD Deprecated notice!
        return $this->compareCategoryName($a, $b);
    }
    public function compareCategoryName($a, $b)
```

Comment (Prestaworks) [reply controllers/admin/AdminModulesController.php:124]: but the old legacy controller is not used any more right? So we can just close this issue and PR?

Comment (boherm) [reply controllers/admin/AdminModulesController.php:124]: Hi @Prestaworks, for now not all legacy controllers are migrated to symfony controllers 😔
I have one dream... complete this long long migration, but it's a lot of work!

Comment (Prestaworks) [reply controllers/admin/AdminModulesController.php:124]: @boherm yes, however :) "checkCategoriesNames" is only used in the constructor of the AdminModulesController.php, removing the code we are talking about

`    // public function checkCategoriesNames($a, $b)
    // {
    //     if ($a['name'] === $this->trans('Other Modules')) {
    //         return true;
    //     }

    //     return (bool) ($a['name'] > $b['name']);
    // }`
the module manager controller is still working, also the entire construct function can be removed from the file and the page in BO is still working (PS 8.1.0).

it's used to sort the categories in list_modules_categories that seems to be used in /themes/default/template/controolers/modules/page.tpl,, however, this tpl file is not used either as the controller for module manager uses twig now. So I think we can close this PR and issue as it's fixed by changing to symfony.

Comment (davidglezz) [reply controllers/admin/AdminModulesController.php:124]: My Suggested change is to rename `checkCategoriesNames` to `compareCategoryName`.
Since `checkCategoriesNames` is public, some no official module could depend on it, then we have to keep the old function as deprecated.
The new comparator, if possible, should be private.
It's only a minor suggestion to improve naming.

Comment (Prestaworks) [reply controllers/admin/AdminModulesController.php:124]: I deleted the file AdminModulesController.php, cleared cache. the module manager page still works fine, I am able to install modules also. I don't think the file is used any more?

Comment (davidglezz) [reply controllers/admin/AdminModulesController.php:124]: Then I think it should be deleted from codebase

Comment (Prestaworks) [reply controllers/admin/AdminModulesController.php:124]: @davidglezz Yes, I agree with you, the file and connected tpl files should be deleted.

**Comment by kpodemski** [comment]
Hello @Prestaworks 
As you noticed, this change is no longer relevant. Sorry for a looooong time waiting 😢

**Pull request closed without merging**
